express-notes-api/server.js

```diff
@@ -50,19 +42,16 @@ app.post('/api/notes', (req, res) => {
     newEntry.id = dataParse.nextId;
     dataParse.notes[dataParse.nextId] = newEntry;
     dataParse.nextId++;
-    console.log(dataParse);
-    res.status(201).json(newEntry);
     const dataStringify = JSON.stringify(dataParse, null, 2);
     writeFile('data.json', dataStringify)
       .then((success) => {
+        res.status(201).json(newEntry);
         console.log('Content was saved');
       })
       .catch((error) => {
-        console.log(error.message);
+        res.status(500).json(error);
```

Comment:
This is OK, but it's generally considered a security hole to return an entire error to a client because the error includes the stack trace, which an attacker might be able to use to learn about your implementation.

The best way to deal with an error is to log it on the server's console and return just the error message:
```
.catch((error) => {
  console.error(error);
  res.status(500).send(error.message);
});
```